django/core/management/commands/populate_memberprofile_affiliations.py

```diff
@@ -0,0 +1,75 @@
+from django.core.management.base import BaseCommand
+from fuzzywuzzy import fuzz
+import re
+import requests
+import time
+import json
+import logging
+
+from core.models import MemberProfile
+
+logger = logging.getLogger(__name__)
+
+class Command(BaseCommand):
+    help = """Migrate data from MemberProfile.institution to MemberProfile.affiliations with
+    an attempt to add more data from ROR database"""
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--cached",
+            type=str
+        )
+
+    def handle(self, *args, **options):
+        # if options["cached"]:
+        #     with open(options["cached"]) as file:
+        #         data = json.load(file)
+
+        for mp in MemberProfile.objects.all().order_by('user_id'):
+            if mp.institution:
+                new_affil = {}
+
+                if mp.institution.name:
+                    new_affil["name"] = mp.institution.name
+                    best_match = self.lookup(mp.institution.name)
+
+                    if best_match:
+                        if (best_match["score"] >= 1.0 and fuzz.partial_ratio(
+                            best_match["organization"]["name"], mp.institution.name) >= 75):
+                            # add ror_id if we have a confident enough match
+                            new_affil["ror_id"] = best_match["organization"]["id"]
+                            # add acronym
+                            if best_match["organization"]["acronyms"]:
+                                new_affil["acronym"] = best_match["organization"]["acronyms"][0]
+                            # add link from the response if the user doesn't have one
+                            if not mp.institution.url and best_match["organization"]["links"]:
+                                new_affil["url"] = best_match["organization"]["links"][0]
+
+                    if mp.institution.url:
```

Comment:
consider nesting this in an else: clause of `if best_match:` so we never override any incoming best_match URL